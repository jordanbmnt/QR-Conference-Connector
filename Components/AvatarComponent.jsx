import {
  Avatar,
  AvatarFallbackText,
  AvatarImage,
  HStack,
  Heading,
  Text,
  VStack,
} from "@gluestack-ui/themed";

export default function AvatarComponent({ name, profilePic, position }) {
  return (
    <VStack space='2xl'>
      <HStack space='md'>
        <Avatar>
          <AvatarFallbackText>{name[0]}</AvatarFallbackText>
          <AvatarImage
            source={{
              uri: profilePic,
            }}
          />
        </Avatar>
        <VStack>
          <Heading color='white' size='sm'>
            {name}
          </Heading>
          {position.length > 0 && (
            <Text size='sm'>{`Position: ${position}`}</Text>
          )}
        </VStack>
      </HStack>
    </VStack>
  );
}
