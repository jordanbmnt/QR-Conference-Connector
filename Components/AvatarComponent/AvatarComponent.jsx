import {
  Avatar,
  AvatarFallbackText,
  AvatarImage,
  HStack,
  Heading,
  Icon,
  VStack,
} from "@gluestack-ui/themed";
import { Dot } from "lucide-react-native";

export default AvatarComponent = ({
  name,
  profilePic,
  status,
  statusColor,
}) => {
  return (
    <VStack space='2xl'>
      <HStack space='md'>
        <Avatar>
          <AvatarFallbackText>{name[0]}</AvatarFallbackText>
          <AvatarImage
            source={{
              uri: profilePic,
            }}
          />
        </Avatar>
        <VStack>
          <Heading color='white' size='md'>
            {name}
          </Heading>
          {status && status.length > 0 && (
            <HStack sx={{ textJustify: "center" }}>
              <Heading color='grey' size='xs'>
                {status}
              </Heading>
              <Icon color={statusColor} as={Dot} size='sm' />
            </HStack>
          )}
        </VStack>
      </HStack>
    </VStack>
  );
};
