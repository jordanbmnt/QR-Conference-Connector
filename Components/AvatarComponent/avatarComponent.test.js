import { config } from "@gluestack-ui/config";
import renderer from "react-test-renderer";
import AvatarComponent from "./AvatarComponent";
import { GluestackUIStyledProvider } from "@gluestack-ui/themed";

jest.useFakeTimers();
describe("AvatarComponent", function () {
  const avatar = renderer
    .create(
      <GluestackUIStyledProvider config={config}>
        <AvatarComponent
          name={"John"}
          profilePic={"profilePic"}
          status={"Alive"}
          statusColor={"orange"}
        />
      </GluestackUIStyledProvider>
    )
    .toJSON();
  const vStack = avatar.children;
  const avatarImage = vStack[0].children[0];
  const avatarInfo = vStack[0].children[1];
  it("should have one child", function () {
    expect(vStack.length).toBe(1);
  });
  it("should have the avatar fallback text of 'J'", function () {
    expect(avatarImage.children[0].children[0]).toBe("J");
  });
  it("should have an image of 'profilePic'", function () {
    const avatarImageInfo = avatarImage.children[1];
    expect(avatarImageInfo.type).toBe("Image");
    expect(avatarImageInfo.props.source).toEqual({ uri: "profilePic" });
  });
  it("should display the name of the parsed content", function () {
    const avatarHeading = avatarInfo.children[0];
    expect(avatarHeading.type).toBe("Text");
    expect(avatarHeading.children[0]).toBe("John");
  });
  it("should have a status of 'Alive'", function () {
    const avatarStatus = avatarInfo.children[1].children[0].children[0];
    expect(avatarStatus).toBe("Alive");
  });
  it("should render avatar component correctly", () => {
    expect(avatar).toMatchSnapshot();
  });
});
