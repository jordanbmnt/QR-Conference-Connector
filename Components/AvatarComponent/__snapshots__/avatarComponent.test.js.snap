// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarComponent should render avatar component correctly 1`] = `
<View
  data-style=""
  dataSet={
    {
      "style": "",
    }
  }
  space="2xl"
  style={
    [
      undefined,
      {},
      {
        "flexDirection": "column",
      },
      {},
      {
        "gap": 24,
      },
      undefined,
    ]
  }
>
  <View
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    space="md"
    style={
      [
        undefined,
        {},
        {
          "flexDirection": "row",
        },
        {},
        {
          "gap": 12,
        },
        undefined,
      ]
    }
  >
    <View
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      size="md"
      style={
        [
          undefined,
          {},
          {
            "alignItems": "center",
            "backgroundColor": "#005DB4",
            "borderRadius": 9999,
            "justifyContent": "center",
            "position": "relative",
          },
          {},
          {
            "height": 48,
            "width": 48,
          },
          undefined,
        ]
      }
    >
      <Text
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        size="md"
        style={
          [
            undefined,
            {},
            {
              "color": "#525252",
              "fontFamily": undefined,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {},
            {
              "fontSize": 16,
              "lineHeight": 22,
            },
            undefined,
            {},
            {
              "color": "#FCFCFC",
              "fontWeight": "600",
              "overflow": "hidden",
              "textTransform": "uppercase",
            },
            {},
            {},
            undefined,
          ]
        }
      >
        J
      </Text>
      <Image
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        onError={[Function]}
        source={
          {
            "uri": "profilePic",
          }
        }
        style={
          [
            undefined,
            {},
            {
              "borderRadius": 9999,
              "height": "100%",
              "position": "absolute",
              "width": "100%",
            },
            {},
            undefined,
          ]
        }
      />
    </View>
    <View
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          {
            "flexDirection": "column",
          },
          {},
          undefined,
        ]
      }
    >
      <Text
        accessibilityRole="header"
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          {
            "color": "white",
            "fontFamily": undefined,
            "fontSize": 18,
            "fontWeight": "700",
            "letterSpacing": -0.2,
            "lineHeight": 24,
            "marginVertical": 0,
          }
        }
      >
        John
      </Text>
      <View
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          [
            undefined,
            {},
            {
              "flexDirection": "row",
            },
            {},
            {
              "textJustify": "center",
            },
            undefined,
          ]
        }
        sx={
          {
            "textJustify": "center",
          }
        }
      >
        <Text
          accessibilityRole="header"
          data-style=""
          dataSet={
            {
              "style": "",
            }
          }
          style={
            {
              "color": "grey",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontWeight": "700",
              "letterSpacing": -0.2,
              "lineHeight": 18,
              "marginVertical": 0,
            }
          }
        >
          Alive
        </Text>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          data-style=""
          dataSet={
            {
              "style": "",
            }
          }
          fill="none"
          focusable={false}
          height={16}
          meetOrSlice={0}
          minX={0}
          minY={0}
          role="image"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              [
                undefined,
                {},
                {
                  "color": "#404040",
                },
                {},
                {
                  "height": 16,
                  "width": 16,
                },
                undefined,
                {},
                {
                  "color": "#404040",
                },
                {},
                {
                  "color": "orange",
                },
                undefined,
              ],
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          tintColor="orange"
          vbHeight={24}
          vbWidth={24}
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          >
            <RNSVGCircle
              cx="12.1"
              cy="12.1"
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              r="1"
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="2"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;
